fish_completions.go

```diff
@@ -141,19 +142,24 @@ function __%[1]s_prepare_completions
 
     set -l nospace (math (math --scale 0 $directive / $shellCompDirectiveNoSpace) %% 2)
     set -l nofiles (math (math --scale 0 $directive / $shellCompDirectiveNoFileComp) %% 2)
+    set -l nomatching (math (math --scale 0 $directive / $shellCompDirectiveNoMatching) %% 2)
 
-    __%[1]s_debug "nospace: $nospace, nofiles: $nofiles"
+    __%[1]s_debug "nospace: $nospace, nofiles: $nofiles, nomatching: $nomatching"
 
     # If we want to prevent a space, or if file completion is NOT disabled,
     # we need to count the number of valid completions.
     # To do so, we will filter on prefix as the completions we have received
     # may not already be filtered so as to allow fish to match on different
     # criteria than the prefix.
     if test $nospace -ne 0; or test $nofiles -eq 0
-        set -l prefix (commandline -t | string escape --style=regex)
-        __%[1]s_debug "prefix: $prefix"
-
-        set -l completions (string match -r -- "^$prefix.*" $__%[1]s_comp_results)
+		if test $nomatching -ne 0
+			set -l completions (string match -r -- "^.*" $__%[1]s_comp_results)
```

Comment:
Doesn't that match everything?  I guess we don't need the `string match` operation in this case.